datasource/netcdf_groundwater.py

```diff
@@ -75,48 +152,120 @@ def line_type_of(self, line_idx):
+        """
+
+        # one example for v2_weir_view
+        # gr.lines
+        # determine wheter v2_weir_view is a lines, nodes or pumps
+        first = getattr(self.gr, get_gr_type(object_type))
+        # gr.lines.filter(content_type='v2_weir', content_pk=1)
+        # we need to get de gridadminname of v2_weir_view (=v2_weir)
+        second = first.filter(content_type=get_gr_name(object_type),
+                              content_pk=object_id)
+        # gr.lines.filter(content_type='v2_channel',content_pk=1).timeseries(
+        # indexes=slice(None))
+        third = second.timeseries(indexes=slice(None))
+        # gr.lines.filter(content_type='v2_channel',content_pk=1).timeseries(
+        # indexes=slice(None)).only('q')
+        fourth = third.only(nc_variable)
+        # gr.lines.filter(content_type='v2_channel',content_pk=1).timeseries(
+        # indexes=slice(None)).only('q').data
+        fifth = fourth.data
+        # fifth is an OrderedDict:
+        # OrderedDict([('q', array([  [ 0.00000000e+00,  0.00000000e+00],
+        #                             [-3.27184124e-10, -7.85903914e-12],
+        # to only get the array we use fifth['q']
+
+        # for now, we just take only the timeseries of the first flowline (
+        # when multiple flowlines are returned
+        # TODO
+        # return all timeseries are do an avg, min, max
+        # e.g. self.vals = np.mean(fifth[nc_variable], axis=1)
+        self.vals = fifth[nc_variable][:, 0]
+
+    def get_timeseries_result_layer(self, object_type, object_id, nc_variable):
+        """
+        -   to get a timeseries using threedigridadmin, we need to call e.g.
+            ga.nodes.filter(id=100).timeseries(indexes=slice(None)).s1
+        -   slice(None) means we get all timesteps.
+        -   get_timeseries_result_layer() gets strings as arguments
+            that we need to parse as atrributes
+        -   using the gridadmin in a python we see that:
+        In [1]: gr.nodes.id
+        Out[1]: array([0, 1, 2, ..., 15601, 15602, 15603])
+        In [2]: len(gr.nodes.id)
+        Out[2]: 15604
+        In [3]: len(set(gr.nodes.id))
+        Out[3]: 15604
+        In [4]: gr.nodes.filter(id=0)
+        this returns empty stuff
+        -   The nodes QgisVectorLayer contains id 1 till 15603 (15603 rows),
+            therefore we can use id to retrieve a timeseries
+        """
+
+        # we need the grdadmin type (e.g. rom flowlines to lines)
+        object_type = get_gr_type(object_type)
+        # one example for nodes
+        # gr.nodes
+        first = getattr(self.gr, object_type)
+        # gr.nodes.filter(id=100)
+        second = first.filter(id=object_id)
+        # gr.nodes.filter(id=100).timeseries(indexes=slice(None))
+        third = second.timeseries(indexes=slice(None))
+        # get the timeserie of the nc_variable
+        self.vals = getattr(third, nc_variable)
+        # flatten numpyarray
+        self.vals = self.vals.flatten()
```

Comment:
No, don't do this. Please program using functions with input and output values: https://upload.wikimedia.org/wikipedia/commons/thumb/3/3b/Function_machine2.svg/220px-Function_machine2.svg.png